worker/worker/parser/category.py

```diff
@@ -16,97 +18,152 @@ def extract_challenge_ids(txt):
     return result
 
 
-def extract_categories(txt):
-    pattern = '<li><a class="submenu.*?" href="fr/Challenges/(.*?)/" '
-    result = re.findall(pattern, txt)
+def extract_categories(r):
+    tree = html.fromstring(r.content)
+    result = tree.xpath('//li/a[starts-with(@class, "submenu")][starts-with(@href, "fr/Challenges")]/@href')
+    result = list(map(lambda x: x.split('/')[2], result))
     if not result:
         raise RootMeParsingError("Could not parse categories.")
     return result
 
 
-def extract_category_logo(txt):
-    txt = txt.replace('\n', '')
-    txt = txt.replace('&nbsp;', '')
-
-    pattern = '<img class=\'vmiddle\' alt="" src="(.*?)" .*?/>'
-    result = re.findall(pattern, txt)
+def extract_category_logo(r):
+    tree = html.fromstring(r.content)
+    result = tree.xpath('//h1/img[@class="vmiddle"][starts-with(@src, "local")]/@src')
     if not result or not result[0]:
         raise RootMeParsingError("Could not parse logo from challenge.")
```

Comment:
Par convention on met pas de majuscule ni de point aux message d'exceptions parce que tu sais pas comment ca va etre log:

Genre par exemple, on pourrait log:

`print(f'parsing error: {err}.')`
ce qui donnerait:
`Parsing error: Could not parse logo from challenge..` (deux points a la fin et une majuscule random au milieu)

Reply:
Oui mais sachant que:
```python
class RootMeParsingError(BaseException):
    pass
```
Est-ce que c'est bien judicieux de mettre un commentaire d'ailleurs 
Pourquoi ne pas tout simplement
```python
raise RootMeParsingError()
```
(si c'est suffisamment explicit)

Reply:
C'est bien d'ajouter du contexte de pourquoi ca a fait de parse